ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs

```diff
@@ -912,7 +929,7 @@ withPeerStateActions PeerStateActionsArguments {
       case res of
         Nothing -> do
           Mux.stopMux pchMux
-          atomically (writeTVar pchPeerStatus PeerCold)
+          atomically (writeTVar pchPeerStatus PeerCooling)
```

Comment:
I can confirm that if the the `peerMonitoringLoop` receives an exception its thread will die but the state won't be correctly updated. I just patched that.

Reply:
> I just patched that.
 
 What did you do exactly?

Reply:
Here: https://github.com/input-output-hk/ouroboros-network/pull/4625/files#diff-7153720053987bcbf5cdfa98695b94b0597288be8a07adc17a9c03445160e7c5R767

Reply:
I am not convinced that's the right thing to do, since if we set the state to `Cold` in an exception handler, why we don't set it to `Cold` in the first place when we throw the exception?

Reply:
If we set the peer to `Cold` right away, then the peer will never be added to `inProgressDemoteToCold` by the monitoring job.

Reply:
If we set the peer to `Cold` right away, then the peer will never be added to `inProgressDemoteToCold` by the monitoring job.

Reply:
Yes you are right!